### DISCUSSION TITLE
Changing the hello message (in the Synergy protocol)

### DISCUSSION THREAD

**Discussion by nbolton** [deskflow/deskflow#7637] [Development]
Proposal from @sithlord48: Change the hello message from "Synergy" to "Barrier"

Let's talk about the pros and cons.

**Pros:**
- It will 'just work' for users with Barrier/IL clients.

**Cons:**
- This will break compatibility with existing Synergy/Deskflow installs (compatibility with Synergy has already been recognized as important by community members, see #7523)
- This will break compatibility with tools such as Wireshark ([detects the Synergy protocol](https://symless.com/synergy/news/how-to-use-wireshark-to-view-synergy-keyboard-network-activity)).

I'll update the pros and cons as we identify more. I'm on holiday right now so I'll give this more thought over the weekend (back in office on Monday 14th).

**Comment by nbolton** [comment]
Only an idea... 

How about we don't have a default? We make the user pick a protocol when adding a client computer in Deskflow.

Which protocol is the client?

(  ) Barrier protocol (for Input Leap and Barrier clients)
(  ) Synergy protocol (for Deskflow and Synergy clients)

That way it's fair to Input Leap and Deskflow.

**Reply by sithlord48** [reply]
Yes, I think that would work.

 - Deskflow Server as described above
    - When a client is added part of the config is to set barrier / synergy mode

 - Deskflow Client 
    - The Client should accept either message and say hello back accordingly
